odoo/modules/module.py

```diff
@@ -276,6 +276,31 @@ def module_manifest(path):
         if os.path.isfile(opj(path, manifest_name)):
             return opj(path, manifest_name)
 
+addons_manifest = {}
+addons_statics = {}
+def load_addons_manifest():
+    """ Load all addons from addons path containing static files and
+    controllers and configure them.  """
+    if addons_manifest:
+        return (addons_manifest, addons_statics)
+
+    for addons_path in odoo.addons.__path__:
+        for module in sorted(os.listdir(str(addons_path))):
+            if module not in addons_manifest:
+                # Deal with the manifest first
+                manifest = read_manifest(addons_path, module)
+                if not manifest or (not manifest.get('installable', True) and 'assets' not in manifest):
+                    continue
+                manifest['addons_path'] = addons_path
+                addons_manifest[module] = manifest
+                # Then deal with the statics
+                path_static = opj(addons_path, module, 'static')
+                if os.path.isdir(path_static):
+                    _logger.debug("Loading %s", module)
+                    addons_statics['/%s/static' % module] = path_static
+
+    return (addons_manifest, addons_statics)
```

Comment:
If nobody is actually calling `load_addons_manifest` to get any information and literally everyone is accessing the attributes directly, why is it returing anything?

If it can and should be called for this, then why isn't it, and why isn't it thread-safe?

Reply:
And if this is supposed to be properly set up on all codepaths, why isn't it just called once during `initialize_sys_path`?

Reply:
Because I didn't know `initialize_sys_path`... I'll have a look. And yes, calling it once at the start would be enough. Here it is called on demand.